rtc_types/src/dh.rs

```diff
@@ -0,0 +1,17 @@
+use crate::EcallResult;
+use sgx_types::*;
+
+pub type SessionRequestResult = EcallResult<sgx_dh_msg1_t, sgx_status_t>;
+pub type ExchangeReportResult = EcallResult<sgx_dh_msg3_t, sgx_status_t>;
+
+impl Default for SessionRequestResult {
+    fn default() -> Self {
+        Self::Err(sgx_status_t::SGX_SUCCESS)
```

Comment:
The reason for using `SGX_SUCCESS` here is because it will always be an invalid state (`Err(SGX_SUCCESS)` should never happen). So the memory will be allocated for the enum, but the state will always be invalid if it were not mutated. This is very opaque, and we should probably do something clearer than this. We need to add a comment for this as a minimum though.

Thanks for catching, any thoughts?

Reply:
Hmm, the case I'm worried about is where something goes wrong, and the default `Err(SGX_SUCCESS)` gets returned here, but then downstream `Err`-handling code actually ends up returning the `SGX_SUCCESS` to code that misinterprets that as an actual success. In other words, it seems risky to introduce an unexpected `SGX_SUCCESS` into downstream error-handling pathways, where the handling of such an unexpected `SGX_SUCCESS` can get very ambiguous (especially if it gets extracted or mapped out from the `Err` container, which a bunch of our pathways do).

So, my gut feel is that we should either:

1. Keep the default `Err` values a status code that will still be a generic SGX-level on its own, such as `SGX_ERROR_UNEXPECTED` or `SGX_ERROR_INVALID_STATE`, to avoid the possibility of any accidental reinterpretation as a success
2. Switch the defaults to the `Ok` case, like:
   ```rust
   Self::Ok(sgx_dh_msg1_t::default())
   ```

I think I'm leaning towards the latter: the defaults for the DH message structs are all zero bytes, which means that if it ever gets accidentally returned, any downstream code that tries to use it should still fail pretty cleanly and expectedly. What do you think?

Reply:
Added in #85